lib/APMTracer.js

```diff
@@ -156,46 +168,62 @@ class APMTracer {
      *         be found in `carrier`
      */
     extract(format, carrier) {
-        if (format === Constants.FORMAT_HTTP_HEADERS ||
-            format === Constants.FORMAT_TEXT_MAP) {
-            return APMTracer._extractHttpAndTextMap(carrier);
+        let spanContext;
+
+        switch (format) {
+        case Constants.FORMAT_HTTP_HEADERS:
+        case Constants.FORMAT_TEXT_MAP:
+            spanContext = APMTracer._extractHttpAndTextMap(carrier);
+            break;
+        default:
+            console.log('Extract unknown format');
+            spanContext = new APMSpanContext();
         }
 
-        console.log('Unsupported format');
-        return new APMSpanContext();
+        return spanContext;
     }
 
     static _extractHttpAndTextMap(carrier) {
-        let spanId;
+        let correlationId = null;
         let traceId;
+        let level;
+        let businessTransaction;
         Object.keys(carrier).forEach((objectKey) => {
-            switch (objectKey.toLowerCase()) {
-            case CARRIER_SPAN_ID:
-                spanId = carrier[objectKey];
+            switch (objectKey.toUpperCase()) {
+            case CARRIER_CORRELATION_ID:
+                correlationId = carrier[objectKey];
                 break;
             case CARRIER_TRACE_ID:
                 traceId = carrier[objectKey];
                 break;
-            case CARRIER_SAMPLED:
+            case CARRIER_BUSINESS_TRANSACTION:
+                businessTransaction = carrier[objectKey];
+                break;
+            case CARRIER_LEVEL:
+                level = carrier[objectKey];
                 break;
-            // ignoring
             default:
-                // error, but ignoring
             }
         });
 
-        if (spanId && traceId) {
-            return new APMSpanContext(spanId, traceId);
+        const spanContext = new APMSpanContext(null, null, null, null);
+        spanContext.setConsumerCorrelationId(correlationId);
+        spanContext.setBusinessTransaction(businessTransaction);
+        spanContext.setLevel(level);
+
+        if (correlationId && traceId) {
+            spanContext._spanId = utils.generateSpanId();
+            spanContext._traceId = traceId;
         }
 
-        // span context not found
-        return new APMSpanContext();
+        return spanContext;
     }
 }
 
 module.exports = {
     APMTracer,
     CARRIER_TRACE_ID,
-    CARRIER_SPAN_ID,
-    CARRIER_SAMPLED,
+    CARRIER_CORRELATION_ID,
+    CARRIER_BUSINESS_TRANSACTION,
```

Comment:
As mentioned above, may be good to change this to just TRANSACTION, and drop the 'business' part.